internal/server/middleware/grpc/middleware.go

```diff
@@ -254,64 +255,71 @@ func AuditUnaryInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
 		// Identity metadata for audit events. We will always include the IP address in the
 		// metadata configuration, and only include the email when it exists from the user logging
 		// into the UI via OIDC.
-		var author string
-		var ipAddress string
+		var (
+			actor  = make(map[string]string)
+			method = "none"
+		)
 
 		md, _ := metadata.FromIncomingContext(ctx)
 		if len(md[ipKey]) > 0 {
-			ipAddress = md[ipKey][0]
+			actor["ip"] = md[ipKey][0]
 		}
 
 		auth := auth.GetAuthenticationFrom(ctx)
 		if auth != nil {
-			author = auth.Metadata[oidcEmailKey]
+			method = strings.ToLower(strings.TrimPrefix(auth.Method.String(), "METHOD_"))
```

Comment:
we could look into adding a `Canonical() string` method or similar that returns this downcased and without the `METHOD_` prefix for all the auth methods as well. But that could come in a future PR

Reply:
@markphelps Agreed can look into that as well.

Reply:
actually, looks like @GeorgeMac already created it! https://github.com/flipt-io/flipt/blob/ca7d91d0cdc01e6d3fb348b33e4989049cbbb74d/internal/config/authentication.go#L217